addons/account/models/account_payment.py

```diff
@@ -101,71 +98,138 @@ class account_register_payments(models.TransientModel):
-        communication = ' '.join([ref for ref in invoices.mapped('reference') if ref])
+        # Look if we are mixin multiple commercial_partner or customer invoices with vendor bills
+        multi = any(inv.commercial_partner_id != invoice_ids[0].commercial_partner_id
+            or MAP_INVOICE_TYPE_PARTNER_TYPE[inv.type] != MAP_INVOICE_TYPE_PARTNER_TYPE[invoice_ids[0].type]
+            for inv in invoice_ids)
+
+        total_amount = self._compute_payment_amount(invoice_ids)
 
         rec.update({
             'amount': abs(total_amount),
-            'currency_id': invoices[0].currency_id.id,
+            'currency_id': invoice_ids[0].currency_id.id,
             'payment_type': total_amount > 0 and 'inbound' or 'outbound',
-            'partner_id': invoices[0].commercial_partner_id.id,
-            'partner_type': MAP_INVOICE_TYPE_PARTNER_TYPE[invoices[0].type],
-            'communication': communication,
+            'partner_id': False if multi else invoice_ids[0].commercial_partner_id.id,
+            'partner_type': False if multi else MAP_INVOICE_TYPE_PARTNER_TYPE[invoice_ids[0].type],
+            'communication': ' '.join([ref for ref in invoice_ids.mapped('reference') if ref]),
+            'invoice_ids': [(6, 0, invoice_ids.ids)],
+            'multi': multi,
         })
         return rec
 
+    @api.model
     def get_payment_vals(self):
-        """ Hook for extension """
+        '''Return the a values dictionary to create a new payment object.
+
+        :return: Payment values as a dict.
+        '''
+        if self.multi:
+            invoice_ids = self._context.get('invoice_ids') or self.invoice_ids
+            amount = self._compute_payment_amount(invoice_ids)
+            payment_type = 'inbound' if amount > 0 else 'outbound'
+            partner_id = invoice_ids[0].commercial_partner_id
+            partner_type = MAP_INVOICE_TYPE_PARTNER_TYPE[invoice_ids[0].type]
+        else:
+            invoice_ids = self.invoice_ids
+            amount = self.amount
+            payment_type = self.payment_type
+            partner_id = self.partner_id
+            partner_type = self.partner_type
         return {
             'journal_id': self.journal_id.id,
             'payment_method_id': self.payment_method_id.id,
             'payment_date': self.payment_date,
             'communication': self.communication,
-            'invoice_ids': [(4, inv.id, None) for inv in self._get_invoices()],
-            'payment_type': self.payment_type,
-            'amount': self.amount,
+            'invoice_ids': [(6, 0, invoice_ids.ids)],
+            'payment_type': payment_type,
+            'amount': abs(amount),
             'currency_id': self.currency_id.id,
-            'partner_id': self.partner_id.id,
-            'partner_type': self.partner_type,
+            'partner_id': partner_id.id,
+            'partner_type': partner_type,
         }
 
     @api.multi
     def create_payment(self):
```

Comment:
Difficult to change the return value of get_payment_vals because it's an hook used in hr_expense_check, account_sepa, account_check_printing, hr_expense and hr_expense_sepa...